doc/sphinx/architecture.md

```diff
@@ -0,0 +1,252 @@
+  - `src/server/frontend_xwayland`
+
+## mirshell
+- The `Shell` is responsible for managing the surfaces that the compositor wants to show.
+- It provides an interface to create, update, and remove these surfaces from the display.
+- As an example, the `WaylandConnector` might ask the `Shell` to create a new surface for it.
+- Directories:
+  - `src/include/server/mir/shell`
+  - `src/server/shell`
+  
+## mirshelldecoration
+- Manages the decoration of *Mir*'s windows
+- Directories:
+  - `src/server/shell/decoration`
+
+## mirscene
+- The `Scene` provides an interface for the `Compositor` to access the list of renderable items
+  - These renderables are derived from the surfaces that were added to the `Shell`.
+  - You can think of the `Scene` as you would think of a "scene graph" in a 3D game engine.
+- Unlike the `Shell`, the `Scene` knows a lot about the Z-order of the surfaces. For this reason, it is also responsible for things like locking the display or sending surfaces to the back of the Z-order.
+- Directories:
+  - `src/include/mir/server/scene`
+  - `src/server/scene`
+
+## mirgraphics
+- A graphics abstraction layer sitting between the compositor and the specific graphics platform that it is using to render.
+- Directories:
+  - `src/include/mir/server/graphics`
+  - `src/server/graphics`
+  
+## mirreport
+- Default logging and reporting facilities
+- Directories:
+  - `src/server/report`
+
+## mirlttng
+- Support for LTTng
+- Directories:
+  - `src/server/report/lttng`
+
+## mirlogging
+- Logging facilities
+- Directories
+  - `src/server/report/logging`
+  
+## mirnullreport
+- A null implementation of the reporting facilities
+- Directories:
+  - `src/server/report/null`
+
+## mirconsole
+- Handles `logind` and virtual-terminal related tasks for the compositor
+- Directories:
+  - `src/server/console`
+
+## mirgl
+- A short list of helpers for GL things
+- Directories:
+  - `src/gl`
+  
+## mirrenderergl
+- The only supported `Renderer` type for now
+- Future renderers will be found alongside it in the `src/renderers` directory
+- Directories:
+  - `src/renderers/gl`
+  
+## mirplatform
+- Graphics and input code that is shared between the platforms found in `src/platforms`
+- Directories
+  - `src/include/platform`
+  - `src/platform`
+
+## mirwayland
+- A subproject used to generate C++ classes from wayland protocol XML files.
+- Directories:
+  - `src/wayland`
+
+## graphics-gbm-kms
+- GBM/KMS display platform for Mir
+- This is the main display platform that one would expect to use
+- Directories:
+  - `src/platforms/gbm-kms`
+
+## graphics-gbm-kms
+- EglStreams/KMS display platform for Mir
+- This is an Nvidia-specific platform
+- Directories:
+  - `src/platforms/eglstreams-kms`
+
+## graphics-x11
+- An X11-based display/input platform
+- Primary platform used for testing
+- Directories:
+  - `src/platforms/x11`
+  
+## graphics-wayland
+- A Wayland-based display/input platform
+- Directories:
+  - `src/platforms/wayland`
```

Comment:
Suddenly we're not prefixing everything with "mir"! :grinning: 

I think it works better to explain what a platform is, and then list examples.

Reply:
I believe I only did this because the final `.so` names aren't prefaced with mir here :joy: But yes, agreed